test/utils.c

```diff
@@ -8,30 +8,42 @@
 
 #include "utils.h"
 
+bool read_sync_flag(struct sync_flag *s) {
+	bool flag;
+	pthread_mutex_lock(&s->mutex);
```

Comment:
The locking here is not necessary, reading a boolean is already an atomic operation. But having a function to read the current flag status is fine.